src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1034,6 +1041,8 @@ private function filterRelativeXPath(string $xpath)
             $crawler->add($domxpath->query($xpath, $node));
         }
 
+        $crawler->cachedNamespaces = $this->cachedNamespaces;
```

Comment:
Rather than copying the cached namespaced a second time to account for changes done in `createDOMXPath`, I'd rather ensure that the changes are done before creating the subcrawler.
This can be done by creating a single `$domxpath` thanks to the invariant that `$node->ownerDocument === $this->document` in Symfony 4+ (that was not enforced in Symfony 3), which could allow us to create a single `$domxpath` before the `createSubCrawler` call and use it in the loop.

Reply:
@simonberger why is this marked as resolved ?